Added tests for models

nicolasbrechet opened the pull request:
Hi,

I've added more tests for Fog::Compute::Exoscale models.

Let me know what you think

Thanks !

nicolasbrechet committed:
Added tests for models

nicolasbrechet committed:
Fixing gems

nicolasbrechet committed:
Fixing gems

nicolasbrechet committed:
Added git location of fog-core

nicolasbrechet committed:
Updated tests for Flavor(s)

nicolasbrechet committed:
Default task is :test

nicolasbrechet committed:
Using lambda syntax to keep compatibility with ruby < 1.9

nicolasbrechet committed:
Updated tests for Image(s)

nicolasbrechet committed:
Updated tests for Job(s)

nicolasbrechet committed:
Updated tests for Zone(s)

nicolasbrechet committed:
Added tests for collections

nicolasbrechet committed:
Updated tests for Security Group(s)

nicolasbrechet committed:
Updated tests for Security Group Rule(s)

nicolasbrechet committed:
Removed useless specs

nicolasbrechet committed:
Updated tests for Server(s)

nicolasbrechet committed:
Typo

nicolasbrechet committed:
Commented failing test

plribeiro3000 commented on .coveralls.yml:
```diff
@@ -0,0 +1,2 @@
+service_name: travis-pro
```
I think you don't need that since is not a pro account.

nicolasbrechet replied:
Thanks! I removed it!

plribeiro3000 commented on lib/fog/exoscale/requests/compute/list_networks.rb:
```diff
@@ -0,0 +1,24 @@
+      class Real
+        # Lists all available networks.
+        #
+        # {CloudStack API Reference}[http://cloudstack.apache.org/docs/api/apidocs-4.4/root_admin/listNetworks.html]
```
CloudStack ?

nicolasbrechet replied:
Exoscale OpenCloud is built on CloudStack.
Since I use code from the cloudstack module in fog, I've added a list of the contributors to the cloudstack module.

plribeiro3000 replied:
Nice. Didn't know about it. =)

nicolasbrechet committed:
Updated coveralls config
Added list of contributors to the Cloudstack module
Added tests for SSH keypairs

nicolasbrechet committed:
Added tests for requests for SSH Keypairs, Security Groups and Servers

nicolasbrechet committed:
Corrected failing tests

plribeiro3000 commented on .coveralls.yml:
```diff
@@ -0,0 +1 @@
+repo_token: XcWZqgPJl8g2Rn7b7zk18azf7WT86i7p0
```
Please don't commit this file.

plribeiro3000 replied:
Don't forget to remove this file and add it to `.gitignore`

nicolasbrechet committed:
9 months later, a bit of work.
- updated ruby version to 2.2.2
- updated tests
- added some mock

nicolasbrechet committed:
Added my latest changes: mocks, new ruby version, corrected tests

nicolasbrechet committed:
Merge branch 'nicolasbrechet-master'

nicolasbrechet committed:
fixed conflict

nicolasbrechet commented:
Hello!
I finally got a bit of time to work on this project...
I've fixed some issues in the tests, and they all pass when testing on my machine...
However I still have an issue with the ENV variables for the API key and Secret key. 
In my fork, some build jobs pass, some don't. What I can see is that in some cases, the environment variables are exported, then the tests pass, and when the env variables are not exported, well... the tests fail.

If you have an idea on how to fix this issue, I'd really appreciate it!

Then I would still have to add fog-exoscale as a runtime_dependency in fog... I also tested that on my machine and it works fine...

nicolasbrechet committed:
Updated travis config

plribeiro3000 commented:
@nicolasbrechet You just pushed your code to both your fork and the main repository.
I've removed the merge commit and now i'm able to see the diff again. Please take a look at your remotes and beware to not do this again please.
If you think its easier you can push it to the main repo but to a branch instead and open the PR to master from it.

Regarding the changes, i will take a look at it later. Thank you for your time and efforts here. :clap:

nicolasbrechet commented:
Oh yes, sorry! Fixed it! Sorry about that!

plribeiro3000 commented on .travis.yml:
```diff
@@ -1,21 +1,27 @@
+
+env:
+  - secure: "LFBE3H1FUpPR4yBoMk0R3xGp6pNR/Dn2/PkoUtdoDiLluNL+2LQVQPFYmfWjPYlbsQa8KY94lVT7pP8lfePkXHEJ4CtjoKGVmQjRG4DTNCrSUTuQiw08tIHBiZJCTZHk8ZoAcQ2ac5Sg9JAxjTpjz2CO77yL1XNzlPlSjZObuAc="
+  - secure: "euxN9HsbO4tn3W3CXUgMxEQJrDK3kACPrRGbqXnYG55BS5gWiM1V7zpjkKSBNlLFYJQidrT63Jrwr4B/RNmZTl5aRoFdEyIqgsfQKrTENxp2c7E5eXNfXExfXx4rYOw2uXeWYrVsyk24XhOjIioQA/zXhMwXEmBXjJd1uAheRw4="
```
Wouldn't it be better to have this configured at Travis where no one have access?

nicolasbrechet replied:
That's what I had... but then the tests always failed. I'm trying to see if that works better but it's probably not the way to do it.

nicolasbrechet committed:
Updated config to avoid extra builds

plribeiro3000 commented on spec/exoscale/requests/security_groups_spec.rb:
```diff
@@ -0,0 +1,107 @@
+        @egress_rule_id = egress_result['jobresult']['securitygroup']['egressrule'][0]['ruleid']
+        
+        @another_sg = @client.list_security_groups['listsecuritygroupsresponse']['securitygroup'].select{|sg| sg['name']==@another_sg['name']}[0]
+      end
```
Wow. What a monster setup. Can't we improve this?
It seems that the code itself does not help much and thus the test reflects it.

Not a blocker, just a comment.

plribeiro3000 commented on spec/exoscale/requests/server_spec.rb:
```diff
@@ -0,0 +1,92 @@
+                                                                      :startPort => '22',
+                                                                      :endPort => '22',
+                                                                      :protocol => 'TCP')['authorizesecuritygroupegressresponse']['jobid']
+      sleep 2 # because it's an async request
```
You should use `Fog.wait_for` here instead. Or in the code itself which would be way better i guess.

plribeiro3000 commented on .rubocop_todo.yml:
```diff
@@ -8,21 +9,50 @@
+    - 'lib/fog/exoscale/requests/compute/list_service_offerings.rb'
+    - 'lib/fog/exoscale/requests/compute/query_async_job_result.rb'
+    - 'lib/fog/exoscale/requests/compute/revoke_security_group_egress.rb'
+    - 'lib/fog/exoscale/requests/compute/revoke_security_group_ingress.rb'
```
I worry that this wont help at all. The code will still be bad. Can we leave as is and let rubocop break until we fix it?

nicolasbrechet committed:
Updated travis config to start the proper rake task (default)

nicolasbrechet committed:
Reverted rubocop_todo to previous state
Removed encrypted env variables as it doesn't help with the authentication issue
Deleted coveralls config file

nicolasbrechet committed:
Simplified tests using Fog.wait_for
